internal/exec/stack_processor_utils.go

```diff
@@ -1741,11 +1740,48 @@ func FindComponentDependenciesLegacy(
 	return unique, nil
 }
 
+// resolveRelativePath checks if a path is relative to the current directory and if so,
+// resolves it relative to the current file's directory. It ensures the resolved path
+// exists within the base path.
+func resolveRelativePath(path string, currentFilePath string) string {
+	if path == "" {
+		return path
+	}
+
+	// Store original path format (using \ or /)
+	originalPathSeparator := "/"
+	if strings.Contains(path, "\\") {
+		originalPathSeparator = "\\"
+	}
+
+	// Convert all paths to use forward slashes for consistency in processing
+	normalizedPath := filepath.ToSlash(path)
+	normalizedCurrentFilePath := filepath.ToSlash(currentFilePath)
+
+	// Determine if this is a relative path by checking if the path starts with "." or ".."
+	parts := strings.Split(normalizedPath, "/")
+	firstElement := filepath.Clean(parts[0])
+	if firstElement == "." || firstElement == ".." {
```

Comment:
Have you already explored `filepath.IsAbs`? If so, what didn't work about it?

Reply:
the path isnt an absolute path since it starts _from_ the stack base path. We could pass that to the function, evaluate that as an absolute path, and then return the original/resolve path

Reply:
that might be better, since we can also check for traversal attacks too

Reply:
> the path isnt an absolute path since it starts from the stack base path.

That's a good point. So we have 2 types of relative paths. Those that begin with `./` or `../` and those that don't.

We already handle those that don't lead with dots. 

You're solving the case when they do. 

Somewhere in the comments of the code, we need to explicitly make this distinction.